File: solutions/javascript/rna-transcription/2/rna-transcription.js

var DnaTranscriber = function () {
};

DnaTranscriber.prototype.toRna = function (dna) {
    dnacode="GCTA";
    rnacode="CGAU";
    var rna =""
    for (var i = 0; i<dna.length; i++) {
        var j = dnacode.indexOf(dna.substring(i,i+1));
        if (j < 0) {
            throw new Error('Invalid input');
        }
        rna = rna + rnacode.substring(j,j+1);
    }  
    return rna;
};

module.exports = DnaTranscriber;

File: solutions/python/rna-transcription/1/rna_transcription.py
def to_rna(dna_strand):
    DNARNA = {"G":"C", "C":"G", "T":"A", "A":"U"}
    rna_strand=""
    for a in dna_strand:
        if a in "GCTA":
            rna_strand += DNARNA[a]
        else:
            raise ValueError("Value Error")
    return(rna_strand)

#print (to_rna("GCTA"))
#print (to_rna("F"))